src/persist-types/src/columnar.rs

```diff
@@ -76,33 +77,45 @@ use crate::Codec;
 /// can be mapped to it's corresponding DataType via [Data::TYPE] and back via
 /// DataType::data_fn.
 pub trait Data: Debug + Send + Sync + Sized + 'static {
-    /// The DataType variant corresponding to this data type.
-    const TYPE: DataType;
+    /// Configuration for the columnar schema of [Self] that is not inferrable
+    /// purely from the type.
+    ///
+    /// For most Data impls, this is unnecessary and set to `()`.
+    type Cfg: ColumnCfg<Self>;
```

Comment:
fair! I partially picked Cfg to keep the three letter names going, but also I didn't come up with any substantially more clear. agreed that Schema is probably not the right thing given the existing usage

Reply:
`ColumnSchema` or some shortening of that seems like it would be sufficiently distinct but still make the connection clear. But I don't feel strongly.

Reply:
i've been letting this sit and i think i'd like to find a way to keep something like this original naming, if possible. the "configuration" for this one initial use of it (dyntable) happens to exactly be a schema, but i sort of think of it as the more general "type of Data impl + config => schema". in fact, a "schema" based naming would arguably be incorrect for all the other Data impls, which set this to (). i'll try to update some doc strings and see if that helps.

Reply:
👍 I find the updated comments here super helpful